File: elixir/06-io-operations/console_output.exs
# IO.puts adds a trailing newline
IO.puts("Line one")
IO.puts("Line two")

# IO.write does NOT add a newline — you control it
IO.write("No newline here... ")
IO.write("still same line\n")

# String interpolation builds formatted messages
name = "Elixir"
version = 1.17
IO.puts("Running #{name} #{version}")

# IO.inspect prints the internal representation of ANY data structure
# and returns its argument, so it drops cleanly into a pipeline
[1, 2, 3]
|> Enum.map(fn x -> x * 2 end)
|> IO.inspect(label: "doubled")

# :io.format is Erlang's printf-style formatter
# ~s = string, ~.2f = float with 2 decimals, ~n = newline
:io.format("~s scored ~.2f~n", ["Elixir", 9.5])

File: julia/06-io-operations/console_output.jl
using Printf

# println adds a trailing newline; print does not
println("println adds a newline")
print("print does not")
print(" — see?\n")

# String interpolation with $
lang = "Julia"
version = 1.11
println("Language: $lang, version $version")

# Formatted output with @printf
@printf("Pi to 4 places: %.4f\n", π)
@printf("Padded integer: |%5d|\n", 42)
@printf("Hex: %x, Octal: %o\n", 255, 8)

# @sprintf returns a formatted string instead of printing it
label = @sprintf("%08.3f", 3.14159)
println("Formatted string: $label")

# show() prints the code representation of a value
show([1, 2, 3])
println()

# Any IO stream can be the target — here, standard error
println(stderr, "This message goes to standard error")
